src/tree.rs

```diff
@@ -62,6 +63,9 @@ impl Anchor {
 
 impl Anchor {
     /// Parses an Orchard anchor from a byte encoding.
+    ///
+    /// We rely upon checking that this is a previously observed anchor, which is
+    /// computed as an output of merklecrh_orchard which performs this validation.
```

Comment:
Similarly, revert this. (It is valid to construct a transaction with an anchor that the node hasn't seen yet, and such a transaction might eventually be valid; in any case we won't be relying on this for type consistency with the spec.)